### PR TITLE
edited imports

### PR DESCRIPTION
**Pull request by nfarabullini**
removal of some direct imports and used `gtx`

### CODE REVIEW

**Comment by nfarabullini** [comment]
launch jenkins spack

**Comment by nfarabullini** [comment]
cscs-ci run default

**Comment by nfarabullini** [comment]
launch jenkins spack

**Comment by nfarabullini** [comment]
cscs-ci run default

**Comment by nfarabullini** [comment]
launch jenkins spack

**Comment by nfarabullini** [comment]
launch jenkins spack

**Comment by nfarabullini** [comment]
cscs-ci run default

**Comment by nfarabullini** [comment]
cscs-ci run default

**Comment by nfarabullini** [comment]
launch jenkins spack

**Comment by nfarabullini** [comment]
cscs-ci run default

**Comment by nfarabullini** [comment]
launch jenkins spack

**Comment by nfarabullini** [comment]
cscs-ci run default

**Comment by nfarabullini** [comment]
cscs-ci run default

**Comment by nfarabullini** [comment]
launch jenkins spack

File: model/atmosphere/diffusion/src/icon4py/model/atmosphere/diffusion/diffusion.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 #
 # Please, refer to the LICENSE file in the root directory.
 # SPDX-License-Identifier: BSD-3-Clause
+import gt4py.next as gtx
```
Comment (jcanton) [inline model/atmosphere/diffusion/src/icon4py/model/atmosphere/diffusion/diffusion.py:8]:
```suggestion
```
This is duplicate at line 18. I thought `pre-commit` should catch this?

File: model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/nh_solve/solve_nonhydro.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
 #
 # Please, refer to the LICENSE file in the root directory.
 # SPDX-License-Identifier: BSD-3-Clause
+import gt4py.next as gtx
```
Comment (jcanton) [inline model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/nh_solve/solve_nonhydro.py:8]: Duplicate at line 14

```suggestion
```

File: model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/state_utils/states.py [hunk 3]
```diff
@@ -5,7 +5,6 @@
 #
 # Please, refer to the LICENSE file in the root directory.
 # SPDX-License-Identifier: BSD-3-Clause
```
Comment (jcanton) [inline model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/state_utils/states.py:7]: This file has a bunch of `gtx.Field` that are missing the `gtx.Dims` (if the secondary objective of this PR is also to fix these things?)

File: tools/src/icon4pytools/py2fgen/template.py [hunk 4]
```diff
@@ -243,7 +243,6 @@ class PythonWrapperGenerator(TemplatedGenerator):
 {% if _this_node.backend == 'GPU' %}import cupy as cp {% endif %}
 from numpy.typing import NDArray
 from gt4py.next.iterator.embedded import np_as_located_field
```
Comment (jcanton) [inline tools/src/icon4pytools/py2fgen/template.py:245]: Is this missing here? `import gt4py.next as gtx`

File: tools/tests/py2fgen/test_codegen.py [hunk 5]
```diff
@@ -327,7 +326,7 @@ def int_array_to_bool_array(int_array: NDArray) -> NDArray:
     return bool_array
 
 @ffi.def_extern()
-def foo_wrapper(one: int32, two: Field[dims.CellDim, dims.KDim], float64], n_Cell: int32, n_K: int32):
+def foo_wrapper(one: int32, two: Field[dims.CellDim, dims.KDim], float64], n_Cell: gtx.int32, n_K: gtx.int32):
```
Comment (jcanton) [inline tools/tests/py2fgen/test_codegen.py:329]: not here?

```suggestion
def foo_wrapper(one: gtx.int32, two: Field[dims.CellDim, dims.KDim], float64], n_Cell: gtx.int32, n_K: gtx.int32):
```

File: tools/tests/py2fgen/test_codegen.py [hunk 6]
```diff
@@ -344,14 +343,13 @@ def foo_wrapper(one: int32, two: Field[dims.CellDim, dims.KDim], float64], n_Cel
     return 0
 
 @ffi.def_extern()
-def bar_wrapper(one: Field[dims.CellDim, dims.KDim], float64], two: int32, n_Cell: int32, n_K: int32):
+def bar_wrapper(one: Field[dims.CellDim, dims.KDim], float64], two: int32, n_Cell: gtx.int32, n_K: gtx.int32):
```
Comment (jcanton) [inline tools/tests/py2fgen/test_codegen.py:346]: not here?

```suggestion
def bar_wrapper(one: Field[dims.CellDim, dims.KDim], float64], two: gtx.int32, n_Cell: gtx.int32, n_K: gtx.int32):
```

File: tools/tests/py2fgen/test_codegen.py [hunk 7]
```diff
@@ -249,7 +249,6 @@ def test_python_wrapper(dummy_plugin):
 import cupy as cp
 from numpy.typing import NDArray
 from gt4py.next.iterator.embedded import np_as_located_field
```
Comment (jcanton) [inline tools/tests/py2fgen/test_codegen.py:251]: Is this necessary here? `import gt4py.next as gtx`

**Comment by jcanton** [comment]
cscs-ci run default

**Comment by jcanton** [comment]
launch jenkins spack

File: tools/src/icon4pytools/py2fgen/template.py (hunk 4 above)
Comment (nfarabullini) [reply tools/src/icon4pytools/py2fgen/template.py:245]: `np_as_located_field` will be deprecated soon so I would leave as is now, plus this is within a docstring and it's trickier to make edits such that tests will pass

File: tools/tests/py2fgen/test_codegen.py (hunk 7 above)
Comment (nfarabullini) [reply tools/tests/py2fgen/test_codegen.py:251]: same as above

**Comment by nfarabullini** [comment]
cscs-ci run default

**Comment by nfarabullini** [comment]
launch jenkins spack

**Review by jcanton — APPROVED** [review]
looks good to me!

**Comment by nfarabullini** [comment]
cscs-ci run default

**Comment by nfarabullini** [comment]
launch jenkins spack

**Comment by nfarabullini** [comment]
cscs-ci run default

**Comment by nfarabullini** [comment]
launch jenkins spack

**Comment by nfarabullini** [comment]
cscs-ci run default

**Pull request merged**